src/NumericalAlgorithms/LinearAlgebra/FindGeneralizedEigenvalues.cpp

```diff
@@ -0,0 +1,99 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/LinearAlgebra/FindGeneralizedEigenvalues.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/Gsl.hpp"
+
+// LAPACK routine to do the generalized eigenvalue problem
+extern "C" {
+extern void dggev_(char*, char*, int*, double*, int*, double*, int*, double*,
+                   double*, double*, double*, int*, double*, int*, double*,
+                   int*, int*);
+}
+
+void find_generalized_eigenvalues(
+    gsl::not_null<DataVector*> eigenvalues_real_part,
+    gsl::not_null<DataVector*> eigenvalues_imaginary_part,
+    gsl::not_null<Matrix*> eigenvectors, const Matrix& A,
```

Comment:
take A and B by value, then get rid of the temporaries `A_for_lapack` and `B_for_lapack`